src/extensions/renderer/base/index.js

```diff
@@ -134,11 +134,13 @@ BRp.notify = function( eventName, eles ){
   }
 
   if(
-    eventName === 'add' || eventName === 'remove'
+    eventName === 'add' 
+    || eventName === 'remove'
     || (eventName === 'move' && cy.hasCompoundNodes())
     || eventName === 'load'
     || eventName === 'zorder'
     || eventName === 'mount'
+    || eventName === 'style'
```

Comment:
Great diagnosis.

If a missed z-sorting is the issue, then this approach would work -- but it is a bit of a shotgun approach. It would undo some performance optimisations, since the z-order would be recalculated every time any style property is changed.  The z-order sorting can be expensive if it happens frequently, and not every property change necessitates a recalculation (e.g. just changing node colour).  This helps keep things like animations on properties like node colour inexpensive.

Defining a `triggersZOrder ` condition in the property spec would be a bit more precise:

https://github.com/cytoscape/cytoscape.js/blob/9e388771139c10cff330dcc586661f8471b000bc/src/style/properties.js#L232

E.g. for the `display` property:

https://github.com/cytoscape/cytoscape.js/blob/9e388771139c10cff330dcc586661f8471b000bc/src/style/properties.js#L237

Aside:  The `text-events` property might also be affected by this bug, since it's analogous to `events`.